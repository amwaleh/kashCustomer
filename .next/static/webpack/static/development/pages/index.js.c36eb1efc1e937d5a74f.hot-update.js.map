{"version":3,"file":"static/webpack/static/development/pages/index.js.c36eb1efc1e937d5a74f.hot-update.js","sources":["webpack:///./pages/index.js"],"sourcesContent":["import React, { Component } from \"react\";\nimport \"semantic-ui-css/semantic.min.css\";\nimport { Form, Message, Grid, Header, Button } from \"semantic-ui-react\";\nimport { createUser, loginUser } from \"../utils/handler\";\nimport AuthService from '../utils/AuthService'\n\nimport \"./styles.css\";\n\n//const auth = new AuthService()\n\nexport default class Page extends Component {\n  state = {\n    data: {},\n    message: \"\",\n    error: \"\",\n    visible: false,\n    showSignUp: false\n  };\n\n  handleChange = e => {\n    const { value, name } = e.target;\n    this.setState(prevState => {\n      return {\n        data: {\n          ...prevState.data,\n          [name]: value\n        }\n      };\n    });\n  };\n\n  handleSubmit = () => {\n    if (this.state.showSignUp) {\n      createUser(this.state.data)\n        .then(response => {\n          this.setState({ message: \"user created\", visible: true });\n        })\n        .catch(e => {\n          this.setState({ error: e.message, visible: true });\n        });\n\n      setTimeout(() => {\n        this.setState({ visible: false });\n      }, 2000);\n    } else {\n      this.propsauth.login(this.state.data)\n    }\n  };\n\n  handleDismiss = () => {\n    this.setState({ visible: false });\n  };\n  toggleForm = () => {\n    this.setState({ showSignUp: !this.state.showSignUp });\n  };\n  render() {\n    const { error, message, visible, showSignUp } = this.state;\n\n    return (\n      <Grid as={\"container\"} className=\"centered\" padded=\"vertically\">\n      <div>\n        {visible && (\n          <Message\n            header={error ? \"Error\" : \"Info\"}\n            content={error || message}\n            color={error ? \"red\" : \"green\"}\n          />\n        )}\n        </div>\n        <Form onSubmit={this.handleSubmit}>\n          <Header as=\"h3\" color=\"teal\">\n            {showSignUp ? \"Sign Up \" : \"Login\"}\n          </Header>\n          <Form.Input\n            name=\"username\"\n            onChange={this.handleChange}\n            type=\"text\"\n            placeholder=\"username\"\n            required\n          />\n          <Form.Input\n            name=\"password\"\n            onChange={this.handleChange}\n            type=\"password\"\n            placeholder=\"password\"\n            required\n          />\n\n          <Button\n            type=\"submit\"\n            content={showSignUp ? \"sign up \" : \"Login\"}\n            color={ showSignUp ? \"blue \" : \"green\" }\n          />\n\n          <Form.Button\n            type=\"button\"\n            basic\n            content={showSignUp ? \"click to login\" : \"click to sign Up\"}\n            onClick={this.toggleForm}\n            className='toogleButton'\n\n          />\n\n        </Form>\n      </Grid>\n    );\n  }\n}\nPage.getInitialProps = async ({ url }) => {\n  const auth = new AuthService()\n\n  if (auth.loggedIn()) {\n    url.replaceTo('/home')\n  }\n  else{\n    return {auth}\n  }\n}\n"],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA;AACA;AACA;AACA;AACA;AAEA;AACA;AAGA;;;;;;;;;;;;;;;;;;AACA;AACA;AACA;AACA;AACA;AACA;AALA;AACA;AAOA;AAAA;AAAA;AAAA;AACA;AACA;AACA;AACA;AADA;AAMA;AACA;AACA;AACA;AACA;AACA;AAEA;AAAA;AAAA;AAAA;AACA;AAEA;AAAA;AAAA;AAAA;AACA;AAEA;AACA;AAAA;AAAA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AACA;AAAA;AAAA;AACA;AACA;AAAA;AACA;AAAA;AAAA;AACA;AACA;;;;;;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAGA;AACA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AACA;AAAA;AAAA;AAAA;AAAA;AAAA;AAGA;AACA;AACA;AAHA;AAAA;AAAA;AAAA;AAAA;AAAA;AAOA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AACA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAIA;AACA;AACA;AACA;AACA;AALA;AAAA;AAAA;AAAA;AAAA;AAAA;AAQA;AACA;AACA;AACA;AACA;AALA;AAAA;AAAA;AAAA;AAAA;AAAA;AASA;AACA;AACA;AAHA;AAAA;AAAA;AAAA;AAAA;AAAA;AAOA;AACA;AACA;AACA;AACA;AALA;AAAA;AAAA;AAAA;AAAA;AAAA;AAYA;;;;AAhGA;AACA;AADA;AACA;AAiGA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AACA;AACA;AAFA;AAAA;AAAA;AAAA;AACA;AAGA;AAJA;AAAA;AACA;AADA;AAAA;AAOA;AAAA;AACA;AARA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AACA;AADA;AAAA;AAAA;AAAA;;;;A","sourceRoot":""}