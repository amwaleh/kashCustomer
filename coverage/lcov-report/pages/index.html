<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for pages</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> pages
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">27.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>22/80</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>64/96</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25.81% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>8/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">24.68% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>19/77</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="create_customer.js"><a href="create_customer.js.html">create_customer.js</a></td>
	<td data-value="26.83" class="pic low"><div class="chart"><div class="cover-fill" style="width: 26%;"></div><div class="cover-empty" style="width:74%;"></div></div></td>
	<td data-value="26.83" class="pct low">26.83%</td>
	<td data-value="41" class="abs low">11/41</td>
	<td data-value="77.78" class="pct medium">77.78%</td>
	<td data-value="72" class="abs medium">56/72</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="14" class="abs low">4/14</td>
	<td data-value="21.05" class="pct low">21.05%</td>
	<td data-value="38" class="abs low">8/38</td>
	</tr>

<tr>
	<td class="file low" data-value="home.js"><a href="home.js.html">home.js</a></td>
	<td data-value="46.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 46%;"></div><div class="cover-empty" style="width:54%;"></div></div></td>
	<td data-value="46.67" class="pct low">46.67%</td>
	<td data-value="15" class="abs low">7/15</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="46.67" class="pct low">46.67%</td>
	<td data-value="15" class="abs low">7/15</td>
	</tr>

<tr>
	<td class="file low" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="24" class="abs low">4/24</td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="22" class="abs low">6/22</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="12" class="abs low">2/12</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="24" class="abs low">4/24</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Mar 02 2019 22:55:34 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
<script src="../block-navigation.js"></script>
</body>
</html>
