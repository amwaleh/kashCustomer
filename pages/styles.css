.centered {
  display: flex;
  flex-flow: column nowrap;
  align-items: center !important;
  justify-content: center;
  height: 100vh;
  flex-flow: column !important;
}

.separator {
  width: 100%;
  height: 10px;
}

.border {
  border: 1px solid black;
  padding: 5px;
}

.wrapper {
  display: flex;
  flex-flow: column nowrap;
  align-items: flex-start;
}

.row {
  display: flex;
  width: 100%;
  margin: 5px 0 5px 0;
}

.row>p {
  word-break: break-word;
}

.row>img {
  margin-right: 15px;
}

.btnwide {
  display: flex;
  justify-content: space-between;
  font-size: 21px;
}

a.toggleButton {
  display: block;
  box-shadow: None !important;
  color: cyan !important;
  margin-top: 20px !important;
}

.formLogin {
  border: 1px solid rgba(255, 250, 250, 0.77);
  width: 300px !important;
  background-color: white;
  padding: 20px 50px !important;
  border-radius: 5px;
}

.submitButtons {
  width: 100%;
  display: flex;
  justify-content: center;
  padding: 20px 0px 50px 0;
}

.fixedMessage {
  position: fixed !important;
  top: 50%;
  bottom: 0;
  left: 30%;
  height: 100px;
  z-index: 999;
  max-width: 70%;
}

.loginBg {
  background: linear-gradient(to bottom, #059a95 15%, #00b5ad 15%, #03444e 67%);
}

.ErrorMsg {
  padding: 20px;
}

.topNavRow {
  padding-bottom: 0 !important;
  margin-bottom: 0 !important;
  z-index: 999;
}

.topNavRow .right {
  margin-right: 100px !important;
  border-left-color: cyan;
}

.sidebarNav {
  padding-top: 12px !important;
  margin-top: 0 !important;
  border-right: 5px solid #000;
  height: 100vh;
}

.sidebarNav .column {
  border-radius: 0 !important;
  padding-right: 0 !important;
}

.sidebarNav .item {
  padding-left: 10px !important;
  margin-bottom: 1px;
  background-color: #FFF !important;
}

.sidebarNav .item:hover {
  background-color: rgba(188, 206, 197, 0.78) !important;
}

.spacer {
  background-color: #00b5ac2d;
  width: 100%;
  height: 200px;
  margin-bottom: 1px;
}

.customerCard {
  margin-bottom: 20px !important;
}

.profileImage {
  filter: grayscale(0.8)
}

.profileActions {
  box-shadow: none !important;
  border: 0 !important;
  height: 17px;
  border-top: 1px solid #e9eae9 !important;
  border-radius: 0 !important;
}

.profileActions .segment {
  padding-top: 5px;
}

a {
  color: rgb(109, 2, 91);
}

.customerDetails {
  box-shadow: none !important;
  border: 0 !important;
  padding: 5px !important;
  margin: 5px 0 !important;
}