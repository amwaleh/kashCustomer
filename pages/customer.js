import React, { Component } from "react";
import "semantic-ui-css/semantic.min.css";
import { Form, Message } from "semantic-ui-react";
import withAuth from '../utils/withAuth'
import { Grid, Container, Header, Segment, Button } from "semantic-ui-react";
import { post, get, put } from "../api";
import './styles.css'
const mainSection = data => [
  ,
  {
    name: "first_name",
    label: "first name",
    required: true,
    value: data && data.first_name
  },
  {
    name: "last_Name",
    label: "last name",
    required: true,
    value: data && data.last_name
  },
  {
    name: "username",
    label: "username",
    required: true,
    value: data && data.username
  },
  {
    name: "email",
    label: "email",
    required: true,
    placeholder: "john@kariuki.com",
    value: data && data.email
  }
];

const billingSection = data => [
  {
    name: "billingFirst_name",
    label: "first name",
    value: data && data.first_name
  },
  {
    name: "billingLast_name",
    label: "last name",
    value: data && data.last_name
  },
  { name: "billingEmail", label: "email", value: data && data.email },
  { name: "billingCity", label: "city", value: data && data.city },
  { name: "billingPhone", label: "phone", value: data && data.phone },
  { name: "billingState", label: "state", value: data && data.state },
  { name: "billingCompany", label: "company", value: data && data.company },
  { name: "billingCountry", label: "country", value: data && data.country },
  { name: "billingPostcode", label: "postcode", value: data && data.postcode },
  {
    name: "billingAddress_1",
    label: "address 1",
    value: data && data.address_1
  },
  {
    name: "billingAddress_2",
    label: "address 2",
    value: data && data.address_2
  }
];

const shippingSection = data => [
  { name: "shippingCity", label: "city", value: data && data.city },
  { name: "shippingState", label: "state", value: data && data.state },
  { name: "shippingCompany", label: "company", value: data && data.company },
  { name: "shippingCountry", label: "country", value: data && data.country },
  { name: "shippingPostCode", label: "postcode", value: data && data.postcode },
  {
    name: "shippingLast_name",
    label: "last name",
    value: data && data.last_name
  },
  {
    name: "shippingAddress_1",
    label: "address 1",
    value: data && data.address_1
  },
  {
    name: "shippingAddress_2",
    label: "address 2",
    value: data && data.address_2
  },
  {
    name: "shippingFirst_name",
    label: "first name",
    value: data && data.first_name
  }
];

const Section = ({ section, onChange }) => (
  <Form.Group as={Grid.Row}>
    {section.map((field, i) => (
      <Grid.Column width={6} key={i}>
        <Form.Input
          label={field.name}
          placeholder={field.placeholder}
          {...field}
          style={{ paddingBottom: "10px" }}
          onChange={e => onChange(e)}
        />
      </Grid.Column>
    ))}
  </Form.Group>
);

export class CustomerForm extends Component {
  state = {
    body: this.props.body,
    error: "",
    visible: false
  };
  static async getInitialProps({ query }) {
    const customer_id = query.id;
    const result = await get(`customers/${customer_id}`);
    const body = JSON.parse(result.body);
    // const {id} = this.props.url.param
    return { body, customer_id };
  }

  onChange = e => {
    const { name, value } = e.target;
    const currentField = name.toLowerCase();
    if (currentField.includes("shipping") || currentField.includes("billing")) {
      const section = currentField.match(/shipping|billing/gi)[0];
      const field = currentField.replace(/shipping|billing/gi, "");
      console.log({ field, section });
      this.setState(prevState => {
        return {
          body: {
            ...prevState.body,
            [section]: {
              ...prevState.body[section],
              [field]: value
            }
          }
        };
      });
    } else {
      this.setState(prevState => {
        return {
          body: { ...prevState.body, [currentField]: value }
        };
      });
    }
  };
  handleSubmit = async e => {
    e.preventDefault();
    try {
      const result = this.props.customer_id
        ? await put(`customers/${this.props.customer_id}`, this.state.body)
        : await post("customers", this.state.body);
      console.log({ result });
      const body = JSON.parse(result.body);
      if (result.statusCode === 201 || result.statusCode === 200) {
        this.props.url.push("/home");
      }
      else{
        this.setState({ error: body.message, visible: true });
      }

    } catch (e) {
      this.setState({ error: e.message, visible: true  });
    }
  };

  render() {
    if (this.state.error) {
      setTimeout(() => {
        this.setState({ visible: false });
      }, 5000);
    }
    const { customer_id} = this.props;
    const { body, error, visible } = this.state;
    return (
      <Grid as={Container} style={{ paddingTop: "30px" }}>
        <Form onSubmit={e => this.handleSubmit(e)}>
          <Grid>

              { visible && <Message
              className='fixedMessage'
                header={ error ? "Error" : "Info" }
                content={ error}
                color={ "red" }
              />}

            <Header as="h3" color="teal">
              Personal info:
            </Header>
            <Section section={mainSection(body)} onChange={this.onChange} />
            <Header as="h3" color="teal">
              Billing:
            </Header>
            <Section
              section={billingSection(body && body.billing)}
              onChange={this.onChange}
            />
            <Header as="h3" color="teal">
              Shipping:
            </Header>
            <Section
              section={shippingSection(body && body.shipping)}
              onChange={this.onChange}
            />
            <Grid.Row >
              <div className='submitButtons'>
            <Form.Button
            size='medium'
            content={ customer_id? "update": "submit"}
            color={ customer_id? "blue": "green"}
            type="submit" />
            </div>
            </Grid.Row>
          </Grid>
        </Form>
      </Grid>
    );
  }
}
export default withAuth(CustomerForm)