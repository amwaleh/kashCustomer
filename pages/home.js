import React, { Component } from "react";
// import { Button, Header, Icon, Image, Modal } from 'semantic-ui-react'
import Head from "next/head";
import Link from "next/link";
import "semantic-ui-css/semantic.min.css";
import "./styles.css";
import withAuth from "../utils/withAuth";
import { get } from "../api";
import {
  Header,
  Container,
  Grid,
  Segment,
  Modal,
  Card,
  Icon,
  Image,
  Pagination,
  Label
} from "semantic-ui-react";

const mapObj = obj => {
  return Object.keys(obj).map((key, i) => (
    <Segment className='customerDetails' key={ i }><b>{key} </b>: {`${obj[key]}`}</Segment>
  ));
};

const CustomerCard = ({ details, onClick }) => (
  <Grid.Column className="customerCard">
    <Card>
      <Card.Content>
        <Card.Content>
          <Image
            size="medium"
            floated="right"
            src={`https://source.unsplash.com/300x300/?smiles ${Math.random()}`}
            className="profileImage"
          />
        </Card.Content>
        <Card.Content>
          <Card.Header>
            {details.first_name} {details.last_name}
          </Card.Header>
          <Card.Meta>
            <span className="date">{details.username}</span>
          </Card.Meta>
        </Card.Content>
        <Card.Content extra>
          <Segment.Group horizontal color="teal" className="profileActions">
            <Modal
              trigger={
                <Segment>
                  <Link>
                    <a>
                      <Icon
                        name="eye"
                        onClick={onClick}
                      />
                      view
                    </a>
                  </Link>
                </Segment>
              }
            >
              <Modal.Header>Customer Details</Modal.Header>
              <Modal.Content image>
                <Image wrapped size="medium" src={details.avatar_url} />
                <Modal.Description>
                  <Label as={Header} color="teal" ribbon>
                    Personal Info
                  </Label>
                  {mapObj(details)}
                  <Label as={Header} color="teal" ribbon>
                    Billing Details
                  </Label>
                  {mapObj(details.billing)}
                  <Label as={Header} color="teal" ribbon>
                    shipping Details
                  </Label>
                  {mapObj(details.shipping)}
                </Modal.Description>
              </Modal.Content>
              <Modal.Actions />
            </Modal>
            <Segment>
              <Link href={{ pathname: "customer", query: { id: details.id } }}>
                <a>
                  <Icon name="edit" />
                  edit
                </a>
              </Link>
            </Segment>
          </Segment.Group>
        </Card.Content>
      </Card.Content>
    </Card>
  </Grid.Column>
);

export class HomePage extends Component {
  state = {
    customers: this.props.customers,
    currentCustomerIndex: null
  };

  handlePageChange = async (e, { activePage }) => {
    const res = await get(`customers?page=${activePage}`);
    const customers = JSON.parse(res.body);
    this.setState({ customers });
  };
  showdetails = index => {
    this.setState({ currentCustomerIndex: this.state.customers[index] });
  };
  render() {
    const { customers } = this.state;
    const totalPages = this.props.headers["x-wp-totalpages"];
    const totalItems = this.props.headers["x-wp-total"];
    return (
      <Grid as={Container}>
        <Grid.Row columns={3}>
          {customers.map((customer, i) => (
            <CustomerCard
              key={i}
              details={customer}
              onClick={() => this.showdetails(i)}
            />
          ))}
        </Grid.Row>
        <Grid.Row>
          <Grid.Column width={10} floated="right">
            {totalItems > 9 && (
              <Pagination
                boundaryRange={0}
                defaultActivePage={1}
                ellipsisItem={null}
                firstItem={null}
                lastItem={null}
                siblingRange={1}
                totalPages={totalPages}
                onPageChange={this.handlePageChange}
              />
            )}
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}

HomePage.getInitialProps = async ({ req }) => {
  const res = await get("customers?per_page=9");
  const data = JSON.parse(res.body);
  console.log(res.headers);
  return { customers: data, headers: res.headers };
};

export default withAuth(HomePage);
