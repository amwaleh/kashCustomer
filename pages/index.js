import React, { Component } from "react";
import "semantic-ui-css/semantic.min.css";
import { Form, Message, Grid, Header, Button } from "semantic-ui-react";
import { createUser, loginUser } from "../utils/handler";
import AuthService from '../utils/AuthService'

import "./styles.css";

const auth = new AuthService()

export default class Page extends Component {
  state = {
    data: {},
    message: "",
    error: "",
    visible: false,
    showSignUp: false
  };
  componentDidMount() {

    if (auth.loggedIn()) {
      this.props.url.replaceTo('/home')
    }
  }
  handleChange = e => {
    const { value, name } = e.target;
    this.setState(prevState => {
      return {
        data: {
          ...prevState.data,
          [name]: value
        }
      };
    });
  };

  handleSubmit = () => {
    if (this.state.showSignUp) {
      createUser(this.state.data)
        .then(response => {
          this.setState({ message: "user created", visible: true });
        })
        .catch(e => {
          this.setState({ error: e.message, visible: true });
        });

      setTimeout(() => {
        this.setState({ visible: false });
      }, 2000);
    } else {

      auth.login(this.state.data)
    }
  };

  handleDismiss = () => {
    this.setState({ visible: false });
  };
  toggleForm = () => {
    this.setState({ showSignUp: !this.state.showSignUp });
  };
  render() {
    const { error, message, visible, showSignUp } = this.state;

    return (
      <Grid as={"container"} className="centered" padded="vertically">
      <div>
        {visible && (
          <Message
            header={error ? "Error" : "Info"}
            content={error || message}
            color={error ? "red" : "green"}
          />
        )}
        </div>
        <Form onSubmit={this.handleSubmit}>
          <Header as="h3" color="teal">
            {showSignUp ? "Sign Up " : "Login"}
          </Header>
          <Form.Input
            name="username"
            onChange={this.handleChange}
            type="text"
            placeholder="username"
            required
          />
          <Form.Input
            name="password"
            onChange={this.handleChange}
            type="password"
            placeholder="password"
            required
          />

          <Button
            type="submit"
            content={showSignUp ? "sign up " : "Login"}
            color={ showSignUp ? "blue " : "green" }
          />

          <Form.Button
            type="button"
            basic
            content={showSignUp ? "click to login" : "click to sign Up"}
            onClick={this.toggleForm}
            className='toogleButton'

          />

        </Form>
      </Grid>
    );
  }
}
