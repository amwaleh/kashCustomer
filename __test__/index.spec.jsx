import * as React from "react";
import { shallow } from "enzyme";
import IndexPage from "../pages/index";
import { CustomerForm } from "../pages/customer";
import { HomePage } from "../pages/home";
const data = {
  email: "john.doe@example.com",
  first_name: "John",
  last_name: "Doe",
  username: "john.doe",
  billing: {
    first_name: "John",
    last_name: "Doe",
    company: "",
    address_1: "969 Market",
    address_2: "",
    city: "San Francisco",
    state: "CA",
    postcode: "94103",
    country: "US",
    email: "john.doe@example.com",
    phone: "(555) 555-5555"
  },
  shipping: {
    first_name: "John",
    last_name: "Doe",
    company: "",
    address_1: "969 Market",
    address_2: "",
    city: "San Francisco",
    state: "CA",
    postcode: "94103",
    country: "US"
  }
};
const customerList = [
  {
    id: 26,
    date_created: "2017-03-21T16:11:14",
    date_created_gmt: "2017-03-21T19:11:14",
    date_modified: "2017-03-21T16:11:16",
    date_modified_gmt: "2017-03-21T19:11:16",
    email: "joao.silva@example.com",
    first_name: "João",
    last_name: "Silva",
    role: "customer",
    username: "joao.silva",
    billing: {
      first_name: "João",
      last_name: "Silva",
      company: "",
      address_1: "Av. Brasil, 432",
      address_2: "",
      city: "Rio de Janeiro",
      state: "RJ",
      postcode: "12345-000",
      country: "BR",
      email: "joao.silva@example.com",
      phone: "(55) 5555-5555"
    },
    shipping: {
      first_name: "João",
      last_name: "Silva",
      company: "",
      address_1: "Av. Brasil, 432",
      address_2: "",
      city: "Rio de Janeiro",
      state: "RJ",
      postcode: "12345-000",
      country: "BR"
    },
    is_paying_customer: false,
    avatar_url:
      "https://secure.gravatar.com/avatar/be7b5febff88a2d947c3289e90cdf017?s=96",
    meta_data: [],
    _links: {
      self: [
        {
          href: "https://example.com/wp-json/wc/v3/customers/26"
        }
      ],
      collection: [
        {
          href: "https://example.com/wp-json/wc/v3/customers"
        }
      ]
    }
  },
  {
    id: 25,
    date_created: "2017-03-21T16:09:28",
    date_created_gmt: "2017-03-21T19:09:28",
    date_modified: "2017-03-21T16:09:30",
    date_modified_gmt: "2017-03-21T19:09:30",
    email: "john.doe@example.com",
    first_name: "John",
    last_name: "Doe",
    role: "customer",
    username: "john.doe",
    billing: {
      first_name: "John",
      last_name: "Doe",
      company: "",
      address_1: "969 Market",
      address_2: "",
      city: "San Francisco",
      state: "CA",
      postcode: "94103",
      country: "US",
      email: "john.doe@example.com",
      phone: "(555) 555-5555"
    },
    shipping: {
      first_name: "John",
      last_name: "Doe",
      company: "",
      address_1: "969 Market",
      address_2: "",
      city: "San Francisco",
      state: "CA",
      postcode: "94103",
      country: "US"
    },
    is_paying_customer: false,
    avatar_url:
      "https://secure.gravatar.com/avatar/8eb1b522f60d11fa897de1dc6351b7e8?s=96",
    meta_data: [],
    _links: {
      self: [
        {
          href: "https://example.com/wp-json/wc/v3/customers/25"
        }
      ],
      collection: [
        {
          href: "https://example.com/wp-json/wc/v3/customers"
        }
      ]
    }
  }
];
describe("Pages", () => {
  describe("Index", () => {
    it("should render without throwing an error", function() {
      const wrapper = shallow(<IndexPage />);
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe("Customers", () => {
    it("should render without throwing an error", function() {
      const wrapper = shallow(<CustomerForm />);
      expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should render Populated fields", function() {
      const wrapper = shallow(<CustomerForm body={data} />);
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe("List Customers", () => {
    it("should render without throwing an error", function() {
      const wrapper = shallow(
        <HomePage
          customers={customerList}
          headers={{ "x-wp-totalpages": 10, "x-wp-total": 12 }}
        />
      );
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
