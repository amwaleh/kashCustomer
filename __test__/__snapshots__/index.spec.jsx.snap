// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Customers should render Populated fields 1`] = `
"<Grid as={[Function: Container]} style={{...}}>
  <Form onSubmit={[Function: onSubmit]} as=\\"form\\">
    <Grid>
      <Header as=\\"h3\\" color=\\"teal\\">
        Personal info:
      </Header>
      <Section section={{...}} onChange={[Function]} />
      <Header as=\\"h3\\" color=\\"teal\\">
        Billing:
      </Header>
      <Section section={{...}} onChange={[Function]} />
      <Header as=\\"h3\\" color=\\"teal\\">
        Shipping:
      </Header>
      <Section section={{...}} onChange={[Function]} />
      <GridRow>
        <div className=\\"submitButtons\\">
          <FormButton size=\\"medium\\" content=\\"submit\\" color=\\"green\\" type=\\"submit\\" as={[Function: FormField]} control={[Function: Button]} />
        </div>
      </GridRow>
    </Grid>
  </Form>
</Grid>"
`;

exports[`Pages Customers should render without throwing an error 1`] = `
"<Grid as={[Function: Container]} style={{...}}>
  <Form onSubmit={[Function: onSubmit]} as=\\"form\\">
    <Grid>
      <Header as=\\"h3\\" color=\\"teal\\">
        Personal info:
      </Header>
      <Section section={{...}} onChange={[Function]} />
      <Header as=\\"h3\\" color=\\"teal\\">
        Billing:
      </Header>
      <Section section={{...}} onChange={[Function]} />
      <Header as=\\"h3\\" color=\\"teal\\">
        Shipping:
      </Header>
      <Section section={{...}} onChange={[Function]} />
      <GridRow>
        <div className=\\"submitButtons\\">
          <FormButton size=\\"medium\\" content=\\"submit\\" color=\\"green\\" type=\\"submit\\" as={[Function: FormField]} control={[Function: Button]} />
        </div>
      </GridRow>
    </Grid>
  </Form>
</Grid>"
`;

exports[`Pages Index should render without throwing an error 1`] = `
"<Grid as={[Function: Container]} className=\\"centered loginBg\\" padded=\\"vertically\\">
  <div className=\\"ErrotMsg\\" />
  <Form onSubmit={[Function]} className=\\"formLogin\\" as=\\"form\\">
    <Header as=\\"h3\\" color=\\"teal\\">
      Login
    </Header>
    <FormInput name=\\"username\\" onChange={[Function]} type=\\"text\\" placeholder=\\"username\\" required={true} as={[Function: FormField]} control={[Function: Input]} />
    <FormInput name=\\"password\\" onChange={[Function]} type=\\"password\\" placeholder=\\"password\\" required={true} as={[Function: FormField]} control={[Function: Input]} />
    <Button type=\\"submit\\" content=\\"Login\\" color=\\"green\\" as=\\"button\\" />
    <a onClick={[Function]} className=\\"toggleButton\\" href=\\"#\\">
      click to sign Up
    </a>
  </Form>
</Grid>"
`;

exports[`Pages List Customers should render without throwing an error 1`] = `
"<Grid as={[Function: Container]}>
  <GridRow columns={3}>
    <CustomerCard details={{...}} onClick={[Function: onClick]} />
    <CustomerCard details={{...}} onClick={[Function: onClick]} />
  </GridRow>
  <GridRow>
    <GridColumn width={10} floated=\\"right\\">
      <Pagination boundaryRange={0} defaultActivePage={1} ellipsisItem={{...}} firstItem={{...}} lastItem={{...}} siblingRange={1} totalPages={10} onPageChange={[Function]} aria-label=\\"Pagination Navigation\\" nextItem={{...}} pageItem={{...}} prevItem={{...}} />
    </GridColumn>
  </GridRow>
</Grid>"
`;
